File: 3sum.cpp
# include <iostream>
# include <vector>
# include <map>
# include <set>
# include <algorithm>

using namespace std;

class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        set<vector<int>> tup_res;
        sort(nums.begin(), nums.end());
        for (int i=0; i<nums.size() - 2; i++){

            for (int j=i+1; j<nums.size() - 1; j++){
                for (int k=j+1; k<nums.size(); k++){
                    if (nums[i] + nums[j] + nums[k] == 0){
                        vector<int> cur_res;
                        cur_res.push_back(nums[i]);
                        cur_res.push_back(nums[j]);
                        cur_res.push_back(nums[k]);
                        tup_res.insert(cur_res);
                    }
                    if (k == nums.size()-1 || nums[k+1] == nums[k]){
                        k += 1;
                    }
                }
                if (j == nums.size()-1 || nums[j+1] == nums[j]){
                    j += 1;
                }
            }
            if (i == nums.size()-1 || nums[i+1] == nums[i]){
                i += 1;
            }
        }
        vector<vector<int>> res_lst(tup_res.begin(), tup_res.end());
        return res_lst;
    }
};

int main(){
    Solution sol;
    vector<int> inp = {-1,0,1,2,-1,-4};
    sol.threeSum(inp);
    return 0;
}

File: 3Sum.py
class Solution:
    def threeSum(self, nums):
        # if len(nums)<3 or len(nums)>3000:
        #     return []
        # for i in range(len(nums)):
        #     if nums[i] < -10**5 or nums[i] > 10**5:
        #         return []

        tups_lst = []
        nums = sorted(nums)

        for i in range(len(nums)):
            for j in range(i+1,len(nums)):
                for k in range(j+1,len(nums)):
                    if nums[i]+nums[j]+nums[k] == 0:
                        onetup = [nums[i],nums[j],nums[k]]
                        is_in = 0
                        for m in range(len(tups_lst)):
                            if set(tups_lst[m]) == set(onetup):
                                is_in = 1
                                break
                        if not is_in:
                            tups_lst.append(onetup)

        return tups_lst


nums = [0,0,0]
s = Solution()
print(s.threeSum(nums))
